### ISSUE TITLE
Missing named properties

### ISSUE THREAD

**Issue by tom-lord** [tom-lord/regexp-examples#14]
The [official ruby documentation](https://ruby-doc.org/core/Regexp.html#class-Regexp-label-Character+Properties) does _not_ include a comprehensive list of all named properties supported by the language. Some examples:

```
/\p{Age=6.0}/
/\p{In Miscellaneous Mathematical Symbols-B}/
/\p{Transport and Map Symbols}/
/\p{Emoji}/ # <-- A valid unicode property name, but NOT(!!) supported by ruby
```

Thankfully, the [onigmo docs](https://github.com/k-takata/Onigmo/blob/master/doc/UnicodeProps.txt) do provide a full list (but not _all_ of these are supported by the latest ruby!)

Possible paths to take:

* Refresh the `db/*.pstore` files with a more comprehensive list
* Has this problem been solved before? Research other [libraries](https://github.com/janlelis/unicode-x).
* Consider directly referencing [RFCs](http://www.unicode.org/Public/UCD/latest/ucd/Blocks.txt) or similar, rather than dynamically generating the lists? (Is this practical?)

Also worth noting:

* This library does not yet "officially" support jruby, because the test suite fails in relation to named properties. (The list supported by this implementation differs to MRI.) Maybe try wrapping the tests in a `rescue SyntaxError`... with caution!! (Define a list of *known errors*; don't just `rescue` blindly.)
* Arbitrary whitespace, underscores and hyphens can be included in unicode property names. This library does not yet account for this.

**Comment by tom-lord** [comment]
Notes to self, after a little more research --

Ruby already directly references the Unicode RFC. For example, for the `\p{In Miscellaneous Mathematical Symbols-B}` example above, we have: https://github.com/ruby/ruby/blob/3628eae2e754a7489feebc6f41371d42d2efcf3c/enc/unicode/11.0.0/name2ctype.h#L34478-L34482

Then, there is this tool in the ruby source code to parse and map property names according to the unicode docs: https://github.com/ruby/ruby/blob/7aaf5b2878210d4df03a84be8d514a553839a5ba/tool/enc-unicode.rb


Lastly, note that ruby defines its uncode version [here](https://github.com/ruby/ruby/blob/1922e6ad65b7a458eeeb84da9f07724b0f1c1145/common.mk#L18-L19). And, due to [this](https://github.com/ruby/ruby/blob/7aaf5b2878210d4df03a84be8d514a553839a5ba/tool/enc-unicode.rb), we can access it at runtime via: `RbConfig::CONFIG['UNICODE_VERSION']`.


---

So in conclusion, I believe it should be possible to access ruby's unicode mappings which are built at compilation; probably with a native C extension. Failing that, the above code should provide enough hints to reproduce the capping generation if necessary.

**Comment by tom-lord** [comment]
Resolved by https://github.com/tom-lord/regexp-examples/pull/40
